### PR TITLE
Update pension claim model context for backwards compatibility

### PR DESCRIPTION
**Pull request by scottsdevelopment**
## Summary

- Defines the kms context model in a backwards compatible way.

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#90163

## Testing done

- enable the module
- submit a pension claim
- in rails console:
  - `SavedClaim::Pension.find([OLD PENSION CLAIM ID]).parsed_form` - no error
  - `Pensions::SavedClaim.find([OLD PENSION CLAIM ID]).parsed_form` - no error
  - `SavedClaim::Pension.find([NEW PENSION CLAIM ID]).parsed_form` - no error
  - `Pensions::SavedClaim.find([NEW PENSION CLAIM ID]).parsed_form` - no error
 - all unit tests are passing

### CODE REVIEW

File: modules/pensions/app/models/pensions/saved_claim.rb [hunk 1]
```diff
@@ -86,5 +86,12 @@ def upload_to_lighthouse(current_user = nil)
 
       Pensions::PensionBenefitIntakeJob.perform_async(id, current_user&.user_account_uuid)
     end
+
+    def kms_encryption_context
```
Comment (wayne-weibel) [inline modules/pensions/app/models/pensions/saved_claim.rb:90]: please add a docstring here indicating what/why this is being done

**Review by wayne-weibel — CHANGES_REQUESTED** [review]
1. Please add an ADR to document the model encryption findings.
2. Add a 'scripts' directory and place the migration script (from the pr description) there
    - the script should be limited to just the 4 claims that need to be corrected
    - update the script to take input so it can be reused without a code change
    - add documentation to the script

**Review by wayne-weibel — CHANGES_REQUESTED** [review]
This does not appear to have worked
5 is a claim saved by/as Pensions::SavedClaim (neither decryption works
4 is a SavedClaim::Pension (only original model decryption works)

image

image

File: modules/pensions/documentation/scripts/migrate_claims_kms_encryption_context.rb
Comment (wayne-weibel) [inline modules/pensions/documentation/scripts/migrate_claims_kms_encryption_context.rb:1]: With the context switching working, this should no longer be needed

**Review by wayne-weibel — CHANGES_REQUESTED** [review]

**Review by wayne-weibel — APPROVED** [review]

**Review by ericboehs — APPROVED** [review]

**Pull request merged**